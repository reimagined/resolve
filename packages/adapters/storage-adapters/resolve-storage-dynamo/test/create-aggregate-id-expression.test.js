import createAggregateIdExpression from '../src/create-aggregate-id-expression'

describe('method "createAggregateIdExpression"', () => {
  test('should return empty expression', () => {
    const {
      conditionExpression,
      attributeNames,
      attributeValues
    } = createAggregateIdExpression({ aggregateIds: undefined })

    expect({
      conditionExpression,
      attributeNames,
      attributeValues
    }).toMatchSnapshot()
  })

  test('should return correctly expression for single aggregateId', () => {
    const {
      conditionExpression,
      attributeNames,
      attributeValues
    } = createAggregateIdExpression({ aggregateIds: ['id1'] })

    expect({
      conditionExpression,
      attributeNames,
      attributeValues
    }).toMatchSnapshot()
  })

  test('should return correctly expression for many aggregatesIds', () => {
    const {
      conditionExpression,
      attributeNames,
      attributeValues
    } = createAggregateIdExpression({ aggregateIds: ['id1', 'id2'] })

    expect({
      conditionExpression,
      attributeNames,
      attributeValues
    }).toMatchSnapshot()
  })
})
