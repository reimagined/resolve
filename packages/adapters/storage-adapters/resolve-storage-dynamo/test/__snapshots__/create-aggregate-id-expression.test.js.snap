// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`method "createAggregateIdExpression" should return correctly expression for many aggregatesIds 1`] = `
Object {
  "attributeNames": Object {
    "#aggregateId": "aggregateId",
  },
  "attributeValues": Object {
    ":aggregateId0": "id1",
    ":aggregateId1": "id2",
  },
  "conditionExpression": "(#aggregateId IN (:aggregateId0,:aggregateId1))",
}
`;

exports[`method "createAggregateIdExpression" should return correctly expression for single aggregateId 1`] = `
Object {
  "attributeNames": Object {
    "#aggregateId": "aggregateId",
  },
  "attributeValues": Object {
    ":aggregateId0": "id1",
  },
  "conditionExpression": "(#aggregateId = :aggregateId0)",
}
`;

exports[`method "createAggregateIdExpression" should return empty expression 1`] = `
Object {
  "attributeNames": Object {},
  "attributeValues": Object {},
  "conditionExpression": "",
}
`;
